src/db.rs

```diff
@@ -2877,12 +2909,168 @@ pub struct LiveFile {
+                    from_cstr(ffi::rocksdb_livefiles_column_family_name(files, i));
+                let name = from_cstr(ffi::rocksdb_livefiles_name(files, i));
+                let directory = from_cstr(ffi::rocksdb_livefiles_directory(files, i));
+                let size = ffi::rocksdb_livefiles_size(files, i);
+                let level = ffi::rocksdb_livefiles_level(files, i);
+
+                // get smallest key inside file
+                let smallest_key = ffi::rocksdb_livefiles_smallestkey(files, i, &mut key_size);
+                let smallest_key = raw_data(smallest_key, key_size);
+
+                // get largest key inside file
+                let largest_key = ffi::rocksdb_livefiles_largestkey(files, i, &mut key_size);
+                let largest_key = raw_data(largest_key, key_size);
+
+                livefiles.push(LiveFile {
+                    column_family_name,
+                    name,
+                    directory,
+                    size,
+                    level,
+                    start_key: smallest_key,
+                    end_key: largest_key,
+                    largest_seqno: ffi::rocksdb_livefiles_largest_seqno(files, i),
+                    smallest_seqno: ffi::rocksdb_livefiles_smallest_seqno(files, i),
+                    num_entries: ffi::rocksdb_livefiles_entries(files, i),
+                    num_deletions: ffi::rocksdb_livefiles_deletions(files, i),
+                });
+            }
+
+            livefiles
+        }
+    }
+}
+
+/// Metadata returned as output from [`Checkpoint::export_column_family`][export_column_family] and
+/// used as input to [`DB::create_column_family_with_import`].
+///
+/// [export_column_family]: crate::checkpoint::Checkpoint::export_column_family
+#[derive(Debug)]
+pub struct ExportImportFilesMetaData {
+    pub(crate) inner: *mut ffi::rocksdb_export_import_files_metadata_t,
+}
+
+impl ExportImportFilesMetaData {
+    pub fn get_db_comparator_name(&self) -> String {
+        unsafe {
+            let c_name =
+                ffi::rocksdb_export_import_files_metadata_get_db_comparator_name(self.inner);
+            let name = from_cstr(c_name);
+            ffi::rocksdb_free(c_name as *mut c_void);
+            name
+        }
+    }
+
+    pub fn set_db_comparator_name(&mut self, name: &str) {
+        let c_name = CString::new(name.as_bytes()).unwrap();
+        unsafe {
+            ffi::rocksdb_export_import_files_metadata_set_db_comparator_name(
+                self.inner,
+                c_name.as_ptr(),
+            );
+        };
+    }
+
+    pub fn get_files(&self) -> Vec<LiveFile> {
+        unsafe {
+            let livefiles_ptr = ffi::rocksdb_export_import_files_metadata_get_files(self.inner);
+            let files = LiveFile::from_rocksdb_livefiles_ptr(livefiles_ptr);
+            ffi::rocksdb_livefiles_destroy(livefiles_ptr);
+            files
+        }
+    }
+
+    pub fn set_files(&mut self, files: &Vec<LiveFile>) {
```

Comment:
If an error occurred after creating the native rocksdb_livefiles_t, it could leak unless explicitly cleaned.

we can use RAII Guard to destroy the container if an early return happens and then forget it only after ownership is transferred to RocksDB.

Reply:
Great suggestion! We could be much more careful here. Coming up.